galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/ITsqHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts;
+
+import javax.validation.constraints.NotNull;
+
+public interface ITsqHandler {
+
+      /**
+       * Set TSQ name
+       * @param queueName TSQ name
+       * @throws TsqException if there is a problem in setting the TSQ name
+       */
+      public void setQName(@NotNull String queueName) throws TsqException;
```

Comment:
I would expect the name to be set when the TSQ was created or injected/got Not after you have a reference to the TSQ-representing object

Reply:
This TSQ handler instance is created for a CICS region.  We can use this instance to read/write/delete multiple TSQs in the CICS region.

Reply:
I think to read/write items from/to a TSQ, you should follow object-oriented design and have something which represents the TSQ from which you read/to which you write.

Having a provider which gets you that instance is OK, but that should be injected.

It's very odd having to set a name to 'target' a handler, then call a method to act on the 'target' that you previously set up.

Reply:
Hi @techcobweb , Does the following TSQ methods look fine to you?

- `String createQueue(String queueName, String data, boolean recoverable);` // If TSQ needs to be created with recoverable option, Returns OK, ALREADY EXISTING, ERROR.
- `void writeQueue(String queueName, String data); `// Non recoverable
- `String readQueue(String queueName, int item);`
- `void deleteQueue(String queueName);`
- `String readQueueNext(String queueName); `// Read next entry in Queue

Reply:
Updated the code as per the new approach using ITsqFactory.